mozdownload/scraper.py

```diff
@@ -118,28 +119,27 @@ def __init__(self, destination=None, platform=None,
             else:
                 self.extension = DEFAULT_FILE_EXTENSIONS[self.platform]
 
-        attempt = 0
-        while True:
-            attempt += 1
-            try:
-                self.get_build_info()
-                break
-            except (errors.NotFoundError, requests.exceptions.RequestException), e:
-                if self.retry_attempts > 0:
-                    # Log only if multiple attempts are requested
-                    self.logger.warning("Build not found: '%s'" % e.message)
-                    self.logger.info('Will retry in %s seconds...' %
-                                     (self.retry_delay))
-                    time.sleep(self.retry_delay)
-                    self.logger.info("Retrying... (attempt %s)" % attempt)
-
-                if attempt >= self.retry_attempts:
-                    if hasattr(e, 'response') and \
-                            e.response.status_code == 404:
-                        message = "Specified build has not been found"
-                        raise errors.NotFoundError(message, e.response.url)
-                    else:
-                        raise
+        self._retry_check_404(self.get_build_info)
+
+    def _retry(self, func, **retry_kwargs):
+        retry_kwargs.setdefault('jitter', 0)
+        retry_kwargs.setdefault('sleeptime', self.retry_delay)
+        retry_kwargs.setdefault('attempts', self.retry_attempts + 1)
```

Comment:
I see. So we could do that in a follow-up issue without any priority. As it is now it's fully ok. Those are just internals.